src/main/java/io/vertx/mqtt/impl/MqttServerImpl.java

```diff
@@ -102,35 +116,7 @@ public synchronized MqttServer exceptionHandler(Handler<Throwable> handler) {
     return this;
   }
 
-  @Override
-  protected void handleMsgReceived(MqttConnection conn, Object msg) {
-    // handling a Netty native MQTT message directly
-    conn.handleMessage(msg);
-  }
-
-  @Override
-  protected MqttConnection createConnection(VertxInternal vertx, ChannelHandlerContext chctx, ContextImpl context, SSLHelper helper, TCPMetrics metrics) {
-    return new MqttConnection(vertx, chctx, vertx.getOrCreateContext(), metrics, this.mqttServerOptions);
-  }
-
-  @Override
-  protected void initChannel(ChannelPipeline pipeline) {
-
-    if (logEnabled) {
-      pipeline.addLast("logging", new LoggingHandler());
-    }
-    pipeline.addLast("mqttEncoder", MqttEncoder.INSTANCE);
-
-    if (this.mqttServerOptions.getMaxMessageSize() > 0) {
-      pipeline.addLast("mqttDecoder", new MqttDecoder(this.mqttServerOptions.getMaxMessageSize()));
-    } else {
-      // max message size not set, so the default from Netty MQTT codec is used
-      pipeline.addLast("mqttDecoder", new MqttDecoder());
-    }
-  }
-
-  @Override
-  protected Object safeObject(Object msg, ByteBufAllocator allocator) {
+  private Object safeObject(Object msg, ByteBufAllocator allocator) {
```

Comment:
I think that now we can get rid of this safeObject method because it's not inherited from a class. We can do the stuff inside the handleMessage of the MqttConnection. I'm going to merge and I already have the update for that to commit on the merged master.